core/common/src/main/java/alluxio/extensions/ExtensionFactoryRegistry.java

```diff
@@ -131,19 +131,18 @@ private synchronized void init() {
     Preconditions.checkArgument(path != null, "path may not be null");
 
     List<T> factories = new ArrayList<>(mFactories);
+    List<T> eligibleFactories = select(path, conf, factories);
+    if (!eligibleFactories.isEmpty()) {
+      return eligibleFactories;
+    }
+
+    factories.clear();
```

Comment:
i don't understand why we can prevent loading native shared lib multiple times by calling clear here.
can you elaborate?

Reply:
> i don't understand why we can prevent loading native shared lib multiple times by calling clear here.
> can you elaborate?

At this line, we did not find any supported factories from the list of factories. So we clear it to avoid to twicely check them wether support or not.